protocols/imap/src/main/java/org/apache/james/imap/api/ImapSessionUtils.java

```diff
@@ -33,13 +37,9 @@ public static MailboxSession getMailboxSession(ImapSession session) {
     }
 
     public static String getUserName(ImapSession imapSession) {
-        final String result;
-        final MailboxSession mailboxSession = getMailboxSession(imapSession);
-        if (mailboxSession == null) {
-            result = null;
-        } else {
-            result = mailboxSession.getUser().getUserName();
-        }
-        return result;
+        Preconditions.checkNotNull(imapSession);
```

Comment:
This _Precondition_ was not in the previous version.
It's not a problem for me, but you may introduce a test for that.

Reply:
It was a previous behavu=iour, I felt better if it's explicit. I'll do.

Reply:
The test is already here : **getUserNameShouldThrowOnNullImapSession**